assets/js/modules/analytics-4/datastore/custom-dimensions-gathering-data.js

```diff
@@ -303,6 +306,90 @@ const baseSelectors = {
 			return false;
 		}
 	),
+
+	/**
+	 * Gets the options for the data availability report.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state           Data store's state.
+	 * @param {string} customDimension Custom dimension slug.
+	 * @return {Object|undefined} Returns the report options if the property is loaded, otherwise undefined.
+	 */
+	getDataAvailabilityReportOptions: createRegistrySelector(
+		( select ) => ( state, customDimension ) => {
+			invariant( customDimension, 'customDimension is required.' );
+
+			const propertyID = select( MODULES_ANALYTICS_4 ).getPropertyID();
+
+			if ( ! propertyID ) {
+				return undefined;
+			}
+
+			const property =
+				select( MODULES_ANALYTICS_4 ).getProperty( propertyID );
```

Comment:
These selectors wouldn't work in a view-only context. The sync shouldn't be attempted in that context anyways so I guess let's make sure that's still the case 😄

Reply:
Understood. For a view-only user, since the property info will be undefined, the custom dimension will be immediately considered to be "gathering data", and hence the report request to check the gathering data state will not be made. So, it looks like we're safe in this context.